include/pybind11/detail/common.h

```diff
@@ -298,6 +298,14 @@
 #    define PYBIND11_TYPE_GUARD_TYPE_HINT "typing_extensions.TypeGuard"
 #endif
 
+#ifndef PYBIND11_DISABLE_NUMERIC_SUPPORTS_HINT
```

Comment:
Naming suggestions:
```
PYBIND11_DISABLE_NUMERIC_SUPPORTS_TYPE_HINTS
PYBIND11_ARG_TYPE_HINT_FLOAT
PYBIND11_ARG_TYPE_HINT_INT
```
(the main idea is to consistently have "type hint" in all macros)